SoftLayer/managers/vs.py

```diff
@@ -818,7 +819,8 @@ def capture(self, instance_id, name, additional_disks=False, notes=None):
         return self.guest.createArchiveTransaction(
             name, disks_to_capture, notes, id=instance_id)
 
-    def upgrade(self, instance_id, cpus=None, memory=None, nic_speed=None, public=True, preset=None):
+    def upgrade(self, instance_id, cpus=None, memory=None, nic_speed=None, public=True, preset=None,
```

Comment:
I dont like having `add` and `disk` here. My suggestion is to have `disks` be an array of disk objects to either upgrade or add, and you can just loop through them to add the prices.

```
disks = [
{'capacity':100, 'number':1}, # upgrade existing disk 1
{'capacity':200, 'number'':-1} # add new disk if the number is -1
]
```

then just remove the `add` parameter.